game-core/src/main/java/games/strategy/engine/framework/startup/ui/pbem/ForumPosterEditorViewModel.java

```diff
@@ -25,12 +43,29 @@
 
   ForumPosterEditorViewModel(final Runnable readyCallback, final GameProperties properties) {
     this.readyCallback = readyCallback;
-    this.forumSelection = (String) properties.get(IForumPoster.NAME);
-    this.topicId = properties.get(IForumPoster.TOPIC_ID, "");
+    setForumSelection((String) properties.get(IForumPoster.NAME));
+    setTopicId(properties.get(IForumPoster.TOPIC_ID, ""));
     this.alsoPostAfterCombatMove = properties.get(IForumPoster.POST_AFTER_COMBAT, false);
     this.attachSaveGameToSummary = properties.get(IForumPoster.INCLUDE_SAVEGAME, false);
   }
 
+  synchronized void setForumSelection(final String forumSelection) {
+    this.forumSelection = forumSelection;
+    forumSelectionValid = !forumSelection.isBlank();
+    viewForumPostButtonEnabled = areFieldsValid();
+    testForumPostButtonEnabled = areFieldsValid();
+    readyCallback.run();
+  }
+
+  synchronized void setTopicId(final String topicId) {
+    this.topicId = topicId;
+    final Integer numericTopicId = Ints.tryParse(topicId);
```

Comment:
Why the `Ints.tryParse` method?
I think I have never seen that before.

Reply:
It is Guava, returns `null` if the string value is not an integer. This avoids using `IntegerParseException` to check validity. It seemed really annoying there is no `isInt(String)` method baked into java yet.